src/services/ipfs-service.js

```diff
@@ -31,40 +31,86 @@ class IpfsService {
     this.mapCache = new MapCache()
   }
 
-  async submitFile(jsonData) {
+  /**
+   * Convert an object to a JSON blob and submit it to IPFS.
+   *
+   * obj {string} - object to save as file
+   */
+  async saveObjAsFile(obj) {
+    let file
+    if (typeof Blob === 'undefined') {
+      file = new Buffer(JSON.stringify(obj))
+    } else {
+      file = new Blob([JSON.stringify(obj)])
+    }
+
+    const ipfsHash = await this.saveFile(file)
+    // Cache the object
+    this.mapCache.set(ipfsHash, obj)
+    return ipfsHash
+  }
+
+  /**
+   * Convert a data URI into a blob and submit it to IPFS.
+   *
+   * dataUri {string} - data uri to convert
+   */
+  async saveDataURIAsFile(dataURI) {
+    // Extract the mime type
+    const mimeString = dataURI.split(',')[0].split(':')[1].split(';')[0]
+    // Decode b64 encoded component
+    const binary = new Buffer(dataURI.split(',')[1], 'base64').toString('binary')
+
+    const buffer = new Array()
+    for (let i = 0; i < binary.length; i++) {
```

Comment:
Something like this would be faster I think:
`const buffer = new Uint8Array([...binary])`